--- ../projectDirectoryExample/CMakePostBuildCommands.cmake	2016-08-17 14:47:53.950414828 -0300
+++ ../projectDirectoryExampleForOpenglIntegration/CMakePostBuildCommands.cmake	2016-08-21 12:12:00.100308210 -0300
@@ -1,80 +1,118 @@
 #the function ADD_RESOURCE_DIRECTORY_TO_DIRECTORY_OF_PROJECT_BUILT
-#copy  ${directory} at "src" to  directory of project built
+#copy  ${resource_orig} to  directory of project built
 FUNCTION(ADD_RESOURCE_DIRECTORY_TO_DIRECTORY_OF_PROJECT_BUILT targetName resource_orig ) 
     #get the last name of directory of resource_orig=================== 
     STRING(REPLACE "/" ";"  transformResource_OrigToAList ${resource_orig})
     LIST(REVERSE transformResource_OrigToAList)
     LIST(GET transformResource_OrigToAList 0 directory)
     #end================================================================
     IF(${CMAKE_GENERATOR} STREQUAL "Unix Makefiles")
         SET(bin_dest "build")
         SET(resource_dest "${bin_dest}/${directory}")
         ADD_CUSTOM_COMMAND (TARGET ${targetName} POST_BUILD
                             COMMAND /bin/sh -c \"if [ ! -d '${resource_dest}' ]; then mkdir ${resource_dest}\; fi\; \"
                             COMMENT "mkdir ${resource_dest}"      
         )
         ADD_CUSTOM_COMMAND (TARGET ${targetName}  POST_BUILD
                             COMMAND cp -rf ${resource_orig}/*  ${resource_dest}
                             COMMENT "cp -rf ${resource_orig}/*  ${resource_dest}"      
         )        
     ENDIF()
     IF(MSVC)
         SET(bin_dest "build")
         ADD_CUSTOM_COMMAND(TARGET ${targetName} POST_BUILD
                            COMMAND  IF NOT EXIST $(Configuration)\\${directory}  mkdir  $(Configuration)\\${directory}
                            COMMENT "IF NOT EXIST $(Configuration)\\${directory}  mkdir  $(Configuration)\\${directory}"
        )
 
        FILE(TO_NATIVE_PATH ${resource_orig} resource_orig_windows)
        ADD_CUSTOM_COMMAND (TARGET ${targetName}  POST_BUILD
                            COMMAND copy \"${resource_orig_windows}\"  $(Configuration)\\${directory}\\
                            COMMENT "copy ${resource_orig_windows}  $(Configuration)\\${directory}\\"      
        )                          
    ENDIF()
    IF(${CMAKE_GENERATOR} STREQUAL "Xcode")
          ADD_CUSTOM_COMMAND (TARGET ${targetName} POST_BUILD
                             COMMAND /bin/sh -c \"if [ ! -d '$\{TARGET_BUILD_DIR\}/${directory}' ]; then mkdir $\{TARGET_BUILD_DIR\}/${directory}\; fi\; \"
                             COMMENT "mkdir $\{TARGET_BUILD_DIR\}/${directory}"      
         )
         ADD_CUSTOM_COMMAND (TARGET ${targetName}  POST_BUILD
                             COMMAND cp -rf ${resource_orig}/*  $\{TARGET_BUILD_DIR\}/${directory}/  
                             COMMENT "cp -rf ${resource_orig}/* $\{TARGET_BUILD_DIR\}/${directory}/"      
         )
     ENDIF()
 ENDFUNCTION(ADD_RESOURCE_DIRECTORY_TO_DIRECTORY_OF_PROJECT_BUILT)
 
 FUNCTION(RUN_POST_BUILD_COMMANDS_DEFAULTS targetName )
 
    IF(${CMAKE_GENERATOR} STREQUAL "Unix Makefiles")
 
         #move complied program to directory build===============================
         GET_PROPERTY(outputFile TARGET ${targetName}   PROPERTY OUTPUT_NAME)
         GET_PROPERTY(outputIsLibrary TARGET ${targetName}   PROPERTY IS_LIBRARY) 
         set(bin_dest "build")        
         ADD_CUSTOM_COMMAND (TARGET ${targetName} POST_BUILD
                             COMMAND /bin/sh -c \"if [ ! -d '${bin_dest}' ]; then mkdir ${bin_dest}\; fi\; \"
                             COMMENT "mkdir ${bin_dest}"      
         )
      
         ADD_CUSTOM_COMMAND (TARGET ${targetName} POST_BUILD
                                 COMMAND /bin/sh -c \"if [ -e 'lib${outputFile}.so' ]; then mv lib${outputFile}.so ${bin_dest}\; fi\; \"
                                 COMMENT "mv lib${outputFile}.so ${bin_dest}/" 
         )
 
         ADD_CUSTOM_COMMAND (TARGET ${targetName} POST_BUILD
                             COMMAND /bin/sh -c \"if [ -e '${outputFile}' ]; then mv ${outputFile} ${bin_dest}\; fi\; \"
                             COMMENT "mv ${outputFile} ${bin_dest}/" 
         )
         #end================================================================
         
     ENDIF()
    
 
 ENDFUNCTION(RUN_POST_BUILD_COMMANDS_DEFAULTS)
 
 FUNCTION(ADD_RESOURCES_DIRECTORIES_TO_DIRECTORY_OF_PROJECT_BUILT targetName parentDir)
     FOREACH(arg ${ARGN})    
         ADD_RESOURCE_DIRECTORY_TO_DIRECTORY_OF_PROJECT_BUILT(${targetName} "${CMAKE_CURRENT_SOURCE_DIR}/${parentDir}/${arg}")
     ENDFOREACH()
 
 ENDFUNCTION(ADD_RESOURCES_DIRECTORIES_TO_DIRECTORY_OF_PROJECT_BUILT)
+
+
+
+#the function ADD_RESOURCE_FILE_TO_DIRECTORY_OF_PROJECT_BUILT
+#copy  ${resource_orig}  to  directory of project built
+FUNCTION(ADD_RESOURCE_FILE_TO_DIRECTORY_OF_PROJECT_BUILT targetName resource_orig ) 
+#get the last name of directory of resource_orig=================== 
+    STRING(REPLACE "/" ";"  transformResource_OrigToAList ${resource_orig})
+    LIST(REVERSE transformResource_OrigToAList)
+    LIST(GET transformResource_OrigToAList 0 directory)
+    #end================================================================
+    IF(${CMAKE_GENERATOR} STREQUAL "Unix Makefiles")
+        SET(bin_dest "build")     
+        SET(resource_dest "${bin_dest}/")        
+        ADD_CUSTOM_COMMAND (TARGET ${targetName}  POST_BUILD
+                            COMMAND cp -rf ${resource_orig}  ${resource_dest}
+                            COMMENT "cp -rf ${resource_orig}  ${resource_dest}"      
+        )        
+    ENDIF()
+    IF(MSVC)        
+       FILE(TO_NATIVE_PATH ${resource_orig} resource_orig_windows)
+       ADD_CUSTOM_COMMAND (TARGET ${targetName}  POST_BUILD
+                           COMMAND copy \"${resource_orig_windows}\"  $(Configuration)\\
+                           COMMENT "copy ${resource_orig_windows}  $(Configuration)\\"      
+       )                          
+   ENDIF()
+   IF(${CMAKE_GENERATOR} STREQUAL "Xcode")         
+        ADD_CUSTOM_COMMAND (TARGET ${targetName}  POST_BUILD
+                            COMMAND cp -rf ${resource_orig}  $\{TARGET_BUILD_DIR\}/  
+                            COMMENT "cp -rf ${resource_orig} $\{TARGET_BUILD_DIR\}/"      
+        )
+   ENDIF()
+ENDFUNCTION(ADD_RESOURCE_FILE_TO_DIRECTORY_OF_PROJECT_BUILT) 
+FUNCTION(ADD_RESOURCES_FILES_TO_DIRECTORY_OF_PROJECT_BUILT targetName parentDir)
+    FOREACH(arg ${ARGN})    
+        ADD_RESOURCE_FILE_TO_DIRECTORY_OF_PROJECT_BUILT(${targetName} "${CMAKE_CURRENT_SOURCE_DIR}/${parentDir}/${arg}")
+    ENDFOREACH()
+ENDFUNCTION(ADD_RESOURCES_FILES_TO_DIRECTORY_OF_PROJECT_BUILT)
\ Falta o caracter nova linha no final do arquivo
