--- ../projectDirectoryExample/src/commonFunctions.h	2016-08-17 15:45:39.978622123 -0300
+++ ../projectDirectoryExampleForOpenglIntegration/src/commonFunctions.h	2016-08-21 12:12:00.112308211 -0300
@@ -1,6 +1,16 @@
 #ifndef commonFunctions_h
 #define  commonFunctions_h
 #include <stdlib.h>
 #include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 char *concatStrings(const char *str1, const char *str2);
+
+#ifdef __cplusplus
+}
+#endif    
+
 #endif
